coderd/httpapi/queryparams.go

```diff
@@ -257,6 +258,23 @@ func (p *QueryParamParser) Strings(vals url.Values, def []string, queryParam str
 	})
 }
 
+func (p *QueryParamParser) JSONStringMap(vals url.Values, def map[string]string, queryParam string) map[string]string {
```

Comment:
We can probably skip the JSON prefix here?

Reply:
I added the prefix because I figured it could be easily confused with a URL-encoded version e.g. `data=key1%3Dvalue1%26key2%3Dvalue2`.

Reply:
Fair! Let's keep it.